### PR TITLE
Documentation: Extend bitstream format chapter

### PR DESCRIPTION
**Pull request by tmichalak**
Working on adding more details to the bitstream format chapter of the Prjxray doc

### CODE REVIEW

**Comment by litghost** [comment]
@mithro Where's the read the docs presubmit?

File: docs/architecture/bitstream_format.rst [hunk 1]
```diff
@@ -1,28 +1,135 @@
...
-    * Type 2 expands word count field to 27 bits by omitting register address
-    * Type 2 must always be proceeded by Type 1 which sets register address
-  * NOP packets are used for inserting required delays
-  * Most registers only accept 1 word of data
-  * Allowed register operations depends on interface used to send packets
-
-    * Writing LOUT via JTAG is treated as a bad command
-    * Single-frame FDRI writes via JTAG fail
+FPGAs are configured with a binary file called the bitstream.
+The bitstream carries the information which logical elements on the fabric should be configured and how in order to implement the target design.
```
Comment (litghost) [inline docs/architecture/bitstream_format.rst]: `the information on which logical elements`

File: docs/architecture/bitstream_format.rst [hunk 2]
```diff
@@ -1,28 +1,135 @@
...
+Moreover, it contains vital information on how to perform the configuration.
+The format of the bitstream is architecture specific, although the formats for devices of the same vendor often share a number of features.
+This chapter covers some details of the bitstream format for FPGA architectures which are currently supported in SymbiFlow's bitstream manipulation tools.
```
Comment (litghost) [inline docs/architecture/bitstream_format.rst]: This sentence is too generic.  This document is about Xilinx bitstreams.

File: docs/architecture/bitstream_format.rst [hunk 3]
```diff
@@ -1,28 +1,135 @@
...
+* CRC - When a packet is received by the device, it automatically updates an internal CRC calculation to include the contents of that packet. A write to the CRC register checks that the calculated CRC matches the expected value written to the register. This CRC check is only used to provide integrity checking of the packet stream, not the configuration memory contents, and are not required for programming. If you are modifying a bitstream, CRC writes can simply be removed instead of recalculating them.
+* Command - Most of the programming sequence is implemented as a state machine that is controlled via one-shot actions. Writes to this register arm an action that, depending on the action requested, may be triggered immediately or delayed until some other condition is met.
+Important Note: During autoincremented frame writes (described later), the current command is rewritten during every autoincrement. This has the effect of rearming the action on every frame written.
+* Frame Address Register (FAR) - Writes to this register set the starting address for the next frame read or write.
```
Comment (litghost) [inline docs/architecture/bitstream_format.rst]: Indent for FAR and FDRI is not correct.  Maybe needs another space or two?

Comment (tmichalak) [reply docs/architecture/bitstream_format.rst]: Fixed.

File: docs/architecture/bitstream_format.rst (hunk 1 above)
Comment (tmichalak) [reply docs/architecture/bitstream_format.rst]: Fixed.

File: docs/architecture/bitstream_format.rst (hunk 2 above)
Comment (tmichalak) [reply docs/architecture/bitstream_format.rst]: Yeah, but it's all true and is inside the Xilinx sub section so the sentence implicitly applies to Xilinx.

**Comment by tmichalak** [comment]
@mithro I think we could merge this if you think the information here is valuable

**Pull request merged**